orchestrator/domain/base.py

```diff
@@ -464,9 +462,18 @@ def get_all_props() -> Generator:
 
         return list(get_all_props())
 
-    def dict(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
-        """Override the dict function to include serializable properties."""
-        attribs = super().dict(**kwargs)
+    @typing_extensions.deprecated("dict() is deprecated and will be removed in the future, use model_dump() instead")
+    def dict(self, *args: Any, **kwargs: Any) -> Dict[str, Any]:
+        warnings.warn(
+            "dict() is deprecated and will be removed in the future, use model_dump() instead",
```

Comment:
should we set a specific date/version

Reply:
This method was an override on BaseModel.dict() which is also merely deprecated, so we shouldn't remove it in my opinion